app/controllers/application_controller.rb

```diff
@@ -35,7 +35,14 @@ def analytics
   end
 
   def create_user_event(event_type, user = current_user)
-    Event.create(user_id: user.id, event_type: event_type)
+    Event.transaction do
+      Event.create!(user_id: user.id, event_type: event_type)
+      # Keep total number of events per user under MAX_EVENTS_PER_USER
+      events = Event.where(user_id: user.id).order('created_at ASC')
+      if events.count > Event::MAX_EVENTS_PER_USER
```

Comment:
I see both points. This is really a kind of data constraint that in an ideal world I would put in the db itself as a stored procedure. You don't want to create w/o also deleting if necessary to keep the count limited.

I suggested the rake task to avoid the ActiveRecord callback, because those can be hard to test sometimes, and because I expect it to be the kind of db task that can happen async. The limit is a convenience to the user and us, not some kind of privacy or legal constraint.